app/Voucher.php

```diff
@@ -22,20 +22,44 @@ class Voucher extends Model
      */
     protected $fillable = [
         'sponsor_id',
-        'redeemer_id',
-        'creditor_id',
+        'trader_id',
         'code',
         'currentstate', // SM_CONFIG looks at this.
     ];
 
     /**
+     * The attributes that should be case to native types.
+     *
+     * @var array
+     */
+    protected $casts = [
+        'sponsor_id' => 'int',
+        'trader_id' => 'int',
+    ];
+
+    /**
      * The attributes that should be hidden for arrays.
      *
      * @var array
      */
     protected $hidden = [
     ];
 
+    /**
+     * Rules for validation.
+     *
+     * @var array
+     */
+    public static $rules = [
+        // Might need to add a 'sometimes' if any required fields can be absent from requests.
+        'trader_id' => ['numeric', 'exists:traders,id'],
```

Comment:
That's fine - it's just the required ones that need a sometimes - if an update can be submitted without including their value